server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -1425,6 +1416,25 @@ public IPAddressVO associateIPToGuestNetwork(long ipId, long networkId, boolean
         }
     }
 
+    protected boolean isSourceNatAvailableForNetwork(Account owner, IPAddressVO ipToAssoc, Network network) {
```

Comment:
Would you mind documenting this method? And explain in a few sentences, what it expect as input, the idea behind it, and then the possible output according to the inputs.

Reply:
Added docus to the method, please check